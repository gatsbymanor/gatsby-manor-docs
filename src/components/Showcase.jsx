import React from 'react';
import Link from 'gatsby-link';
import classNames from 'classnames';

import "../styles/_showcase.scss";

import massivelyImg from '../images/massively.png';
import dimensionImg from '../images/dimension.png';
import fortyImg from '../images/forty.png';
import identityImg from '../images/identity.png';
import etherealImg from '../images/ethereal.png';
import parallelismImg from '../images/parallelism.png'
import lensImg from '../images/lens.png'
import heliosImg from '../images/helios.png'



const Showcase = (props) => {
  const { data, } = props;

  let ActionButton = null;

  if(!!data.lead.callToAction) {
    ActionButton = (
      <p className="showcase--lead">
        <Link to={data.lead.callToAction.path} className="showcase--lead--action -primary">
          {data.lead.callToAction.text}
        </Link>
      </p>
    );
  }

  let showcaseClassses = classNames({
    'showcase': true,
  });


  return (
    <div className={showcaseClassses}>
      <div className="showcase--images">
        <div className="showcase--gradient">
          <img className="showcase--img" src={lensImg} />
          <img className="showcase--img" src={parallelismImg} />
          <img className="showcase--img" src={heliosImg} />
          <img className="showcase--img" src={identityImg} />
        </div>
      </div>

      <div className="showcase--cta">
        <span className="showcase--title">{data.lead.title}</span>
        <p className="showcase--lead">
          <span>{data.lead.partOne}</span>
        </p>
        {ActionButton}
      </div>
    </div>
  );
};

export default Showcase;
