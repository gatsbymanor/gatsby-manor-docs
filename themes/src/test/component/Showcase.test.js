import React from 'react';
import Showcase from '../../src/components/Showcase';

describe('<Showcase />', () => {
  const testData = {
    lead: {
      title: 'Theme Gallery',
      partOne: 'Developers deserve beautiful designs.',
      partTwo: 'Find a theme. Let Jay do the rest.',
      callToAction: {
        path: '/hello',
        text: 'hello world',
      },
    }
  };

  const component = shallow(
    <Showcase
      data={testData}
      images={testData} />
  );

  it('should render', () => {
    expect(component).toMatchSnapshot();
  });

});
