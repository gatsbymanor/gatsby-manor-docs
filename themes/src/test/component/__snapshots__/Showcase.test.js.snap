// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Showcase /> should render 1`] = `
<div
  className="showcase"
>
  <div
    className="showcase--images"
  >
    <div
      className="showcase--gradient"
    >
      <Image
        alt="title theme"
        className="showcase--img"
        fadeIn={true}
        title="Theme image"
      />
    </div>
  </div>
  <div
    className="showcase--cta"
  >
    <span
      className="showcase--title"
    >
      Theme Gallery
    </span>
    <p
      className="showcase--lead"
    >
      <span>
        Developers deserve beautiful designs.
      </span>
    </p>
    <p
      className="showcase--lead"
    >
      <GatsbyLink
        className="showcase--lead--action -primary"
        to="/hello"
      >
        hello world
      </GatsbyLink>
    </p>
  </div>
</div>
`;
